src/components/flyout/flyout.tsx

```diff
@@ -263,35 +254,33 @@ export const EuiFlyout = forwardRef(
     };
 
     let newStyle;
-    let widthClassName;
-    let sizeClassName;
 
     // Setting max-width
-    if (maxWidth === true) {
-      widthClassName = 'euiFlyout--maxWidth-default';
-    } else if (maxWidth !== false) {
+    if (maxWidth !== false) {
       const value = typeof maxWidth === 'number' ? `${maxWidth}px` : maxWidth;
       newStyle = { ...style, maxWidth: value };
     }
 
     // Setting size
-    if (isEuiFlyoutSizeNamed(size)) {
-      sizeClassName = sizeToClassNameMap[size];
-    } else if (newStyle) {
+    if (newStyle) {
       newStyle.width = size;
     } else {
       newStyle = { ...style, width: size };
     }
 
-    const classes = classnames(
-      'euiFlyout',
-      typeToClassNameMap[type as _EuiFlyoutType],
-      sideToClassNameMap[side as _EuiFlyoutSide],
-      sizeClassName,
-      paddingSizeToClassNameMap[paddingSize as _EuiFlyoutPaddingSize],
-      widthClassName,
-      className
-    );
+    const euiTheme = useEuiTheme();
+    const styles = euiFlyoutStyles(euiTheme, paddingSize);
+
+    const cssStyles = [
+      styles.euiFlyout,
+      styles.paddingSizes[paddingSize],
+      side === 'left' && type === 'push' && styles['push--left'],
+      isEuiFlyoutSizeNamed(size) && styles[`flyoutSize--${size}`],
```

Comment:
see above re `--` modifers. let's also use an object for flyoutSize instead of a modifier. The end className result becomes a bit more opaque, but it's something we've already done in previous Emotion conversions so I'm not overly worried about it

```suggestion
      side === 'left' && type === 'push' && styles.pushLeft,
      isEuiFlyoutSizeNamed(size) && styles.flyoutSizes[size],
```